neal/sampler.py

```diff
@@ -346,6 +356,33 @@ def sample(self, bqm, beta_range=None, num_reads=None, num_sweeps=1000,
             num_sweeps_per_beta, beta_schedule,
             seed, numpy_initial_states, interrupt_function)
 
+        # change the format according to answer_mode
+        if answer_mode == 'histogram':
```

Comment:
I believe that [sampleset.aggregate()](https://docs.ocean.dwavesys.com/projects/dimod/en/latest/reference/generated/dimod.SampleSet.aggregate.html#dimod.SampleSet.aggregate) should give you this behaviour.